src/main/java/com/marcosoft/kartracingapi/service/CrudService.java

```diff
@@ -100,6 +149,15 @@ public void deleteRoundPilot(RoundPilotEntity roundPilot){
         roundPilotEntityRepository.delete(roundPilot);
     }
 
+    public List<RoundPilotEntity> getRoundPilotList(){
+        return (List<RoundPilotEntity>) roundPilotEntityRepository.findAll();
+    }
+
+    public RoundPilotEntity getSingleRoundPilot(Integer roundPilotId){
+        Optional<RoundPilotEntity> roundPilot = roundPilotEntityRepository.findById(roundPilotId);
+        return roundPilot.orElse(null);
```

Comment:
sugestion: Don 't return null. 
Return the optional and then map the optional to return the http status
Example
```
service.getSingleRoundPilot(roundPilotId)
                .map(ResponseEntity::ok)
                .orElse(ResponseEntity.status(HttpStatus.NOT_FOUND).build());
```